src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -44,20 +44,29 @@ public static object ConvertFromJson(string input, out ErrorRecord error)
             }
             error = null;
 #if CORECLR
-            object obj = JsonConvert.DeserializeObject(input, new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.None, MaxDepth = 1024 });
-
-            // JObject is a IDictionary
-            if (obj is JObject)
+            object obj = null;
+            try
             {
-                var dictionary = obj as JObject;
-                obj = PopulateFromJDictionary(dictionary, out error);
-            }
+                obj = JsonConvert.DeserializeObject(input, new JsonSerializerSettings() { TypeNameHandling = TypeNameHandling.None, MaxDepth = 1024 });
 
-            // JArray is a collection
-            else if (obj is JArray)
+                // JObject is a IDictionary
+                if (obj is JObject)
+                {
+                    var dictionary = obj as JObject;
+                    obj = PopulateFromJDictionary(dictionary, out error);
+                }
+
+                // JArray is a collection
+                else if (obj is JArray)
```

Comment:
Ok, I'm going to refactor this expression too. However, it will increase the nesting.